server/test_registry.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"github.com/google/trillian/storage"
+)
+
+type GetLogStorageFunc func() (storage.LogStorage, error)
+type GetMapStorageFunc func() (storage.MapStorage, error)
+
+// testRegistry implements extension.Registry.
+// It delegates its implementation to its member funcs.
+// It is intended as a stub for testing.
+type testRegistry struct {
```

Comment:
I wonder if we could get away with just using a MockExtensionRegistry in the tests above.

Feel free to push back on this.